UnicornRender/video/include/unicorn/video/CameraProjection.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_CAMERAPROJECTION_HPP
+#define UNICORN_VIDEO_CAMERAPROJECTION_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+#include <unicorn/system/Window.hpp>
+#include <glm/glm.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+
+/** @brief Camera projection abstract class for orthographical and perspetive projections */
+class CameraProjection
+{
+public:
+    /** @brief Updates camera projection */
+    virtual void UpdateProjection() = 0;
+protected:
+    CameraProjection(system::Window* window, glm::mat4& cameraProj);
```

Comment:
So do you connect to any signals? Either way you should disclose if projection matrix will be updated when window is resized so that user doesn't call `UpdateProjection` manually